src/math_tools/central_fdm_partial_derivatives_with_boundary_values.hpp

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ddc/uniform_point_sampling.hpp"
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+#include "ipartial_derivative.hpp"
+
+
+/**
+ * @brief A class which implements a partial derivative operator
+ * using a finite differences calculation of order two. A decentered
+ * scheme is used at the boundary, whereas centred finite difference
+ * are used inside the domain.
+ *
+ * @tparam IdxRangeFull The index range of the field on which the operator acts 
+ * (with all dimensions, batched and dimension of interest).
+ * @tparam DerivativeDimension The dimension on which the partial derivative is calculated.
+ */
+template <class IdxRangeFull, class DerivativeDimension>
+class CentralFDMPartialDerivativeWithBValue
+    : public IPartialDerivative<IdxRangeFull, DerivativeDimension>
+{
+private:
+    using base_type = IPartialDerivative<IdxRangeFull, DerivativeDimension>;
+
+    /// The type of the field to be differentiated.
+    using DFieldMemType = DFieldMem<IdxRangeFull>;
+
+    /// The type of a reference to the field to be differentiated.
+    using typename base_type::DFieldType;
+
+    /// The type of a constant reference to the field to be differentiated.
+    using typename base_type::DConstFieldType;
+
+    /// The index range of the dimension on which the partial derivative is calculated.
+    using typename base_type::IdxRangeDeriv;
+
+    /// The index range of all dimensions except DerivativeDimension.
+    using typename base_type::IdxRangeBatch;
```

Comment:
FYI you only need to give detailed comments for public attributes.

That said I'm not sure why these are private. I'm not sure nvidia likes it when the types used in the function prototypes are private